### ISSUE TITLE
Add attachments retrieval

### ISSUE THREAD

**Issue by Frank995** [pydiscourse/pydiscourse#95]
Hi all.

Just started working with this library and I noticed that there is no functionality to retrieve attachments from posts, or upload one. Is it on the roadmap?

**Comment by bennylope** [comment]
It could be! Interested in contributing it?

**Comment by Frank995** [comment]
Yes I would. Currently I'm using a dirty regex to get the attachment url form the post HTML an then send the request manually by callig "_get". Would you have any advice or a reference for how to retrieve them more cleanly?

**Comment by bennylope** [comment]
> Would you have any advice or a reference for how to retrieve them more cleanly?

No idea!

And there doesn't seem to be any support for this in the official [Ruby API client](https://github.com/search?q=repo%3Adiscourse%2Fdiscourse_api%20attachment&type=code).

That being said, I'd be less shy about using regex. The Discourse app itself uses a CSS selector to identify attachments from posts!

https://github.com/discourse/discourse/blob/d91456fd53c4bb462058ecdd676e980c6b2a1dd6/app/models/post_analyzer.rb#L73

A method returning a list of attachments from an individual post sounds useful; wouldn't worry too much if it uses regex if it works.

**Comment by goetzk** [comment]
There are several ways to upload files documented in the API docs (https://docs.discourse.org/#tag/Uploadshttps://docs.discourse.org/#tag/Uploads) but I don't see anything about specifically listing uploads. Weird!

I searched for a while on https://meta.discourse.org/ and didn't find anything there either, so it really looks like pulling data out via regex is the "supported" way.

**Comment by Frank995** [comment]
Got it, that's very weird. I'm working on it!

**Comment by Frank995** [comment]
I wanted to push a branch but I got permission denied

**Comment by goetzk** [comment]
> I wanted to push a branch but I got permission denied

You need to use the "fork" button on GitHub to create a copy then clone that copy to your computer.
On mobile so can't look up the docs, hopefully that gets you started.

**Comment by Frank995** [comment]
I did push one, open to review and suggestions

**Comment by goetzk** [comment]
I've added a review - thanks for your submission!
